client/k8s/k8s_test.go

```diff
@@ -0,0 +1,263 @@
+							CAFile:   "/var/run/kubernetes/server-ca.crt",
+						},
+					}, nil
+				},
+			},
+			expectedHost:     "http://in-cluster-host",
+			expectedCertFile: "/var/run/kubernetes/client-admin.crt",
+			expectedKeyFile:  "/var/run/kubernetes/client-admin.key",
+			expectedCAFile:   "/var/run/kubernetes/server-ca.crt",
+		},
+		{
+			name: "Do not specify anything while using in-cluster config. It should return an error.",
+			config: Config{
+				InCluster: true,
+				inClusterConfigProvider: func() (*rest.Config, error) {
+					return nil, fmt.Errorf("No in-cluster config")
+				},
+			},
+			expectedError: true,
+		},
+		{
+			name: "Specify both in-cluster config and CLI config. It should return the in-cluster config. Use basic auth.",
+			config: Config{
+				InCluster: true,
+				Host:      "http://host-from-cli",
+				Username:  "cli-user",
+				Password:  "cli-password",
+				inClusterConfigProvider: func() (*rest.Config, error) {
+					return &rest.Config{
+						Host:     "http://in-cluster-host",
+						Username: "in-cluster-user",
+						Password: "in-cluster-password",
+					}, nil
+				},
+			},
+			expectedHost:     "http://in-cluster-host",
+			expectedUsername: "in-cluster-user",
+			expectedPassword: "in-cluster-password",
+		},
+		{
+			name: "Specify both in-cluster config and CLI config. It should return the in-cluster config. Use token auth.",
+			config: Config{
+				InCluster:   true,
+				Host:        "http://host-from-cli",
+				BearerToken: "20eeff4fea764a6020e767f224ffb2a0ea3fc48bf11e0aadf99c3ee7092e29bd",
+				inClusterConfigProvider: func() (*rest.Config, error) {
+					return &rest.Config{
+						Host:        "http://in-cluster-host",
+						BearerToken: "53b3d9d43417971220107dafd2f72def8bd26d2ba93ad4b1eccab98e89b7371f",
+					}, nil
+				},
+			},
+			expectedHost:        "http://in-cluster-host",
+			expectedBearerToken: "53b3d9d43417971220107dafd2f72def8bd26d2ba93ad4b1eccab98e89b7371f",
+		},
+		{
+			name: "Specify both in-cluster config and CLI config. It should return the in-cluster config. Use cert auth files.",
+			config: Config{
+				InCluster: true,
+				Host:      "http://host-from-cli",
+				TLSClientConfig: TLSClientConfig{
+					CertFile: "/var/run/kubernetes/client-cli.crt",
+					KeyFile:  "/var/run/kubernetes/client-cli.key",
+					CAFile:   "/var/run/kubernetes/server-ca-cli.crt",
+				},
+				inClusterConfigProvider: func() (*rest.Config, error) {
+					return &rest.Config{
+						Host: "http://in-cluster-host",
+						TLSClientConfig: rest.TLSClientConfig{
+							CertFile: "/var/run/kubernetes/client-admin.crt",
+							KeyFile:  "/var/run/kubernetes/client-admin.key",
+							CAFile:   "/var/run/kubernetes/server-ca.crt",
+						},
+					}, nil
+				},
+			},
+			expectedHost:     "http://in-cluster-host",
+			expectedCertFile: "/var/run/kubernetes/client-admin.crt",
+			expectedKeyFile:  "/var/run/kubernetes/client-admin.key",
+			expectedCAFile:   "/var/run/kubernetes/server-ca.crt",
```

Comment:
I almost always find the naming of certificate files and variables around them very confusing. We tend to agree on some naming scheme in our code. Certificate files are named `ca.pem`, `crt.pem`, `key.pem`. Variables dealing with these concepts align to that. So I would like to rename `expectedCertFile` to `expectedCrtFile`. Below is also this code. 

```
+			k8sTlsClientConfig := k8sclient.TLSClientConfig{ 
+				CertFile: Flags.Kubernetes.TLS.Client.CertificateFile,
+				KeyFile:  Flags.Kubernetes.TLS.Client.KeyFile,
+				CAFile:   Flags.Kubernetes.TLS.Authority.CertificateFile,
+			}
```

I would like to rename `CertFile` to `CrtFile` and `Flags.Kubernetes.TLS.Client.CertificateFile` to `Flags.Kubernetes.TLS.CrtFile`.

I would also like to rename `CAFile:   Flags.Kubernetes.TLS.Authority.CertificateFile` to `CAFile:   Flags.Kubernetes.TLS.CaFile`.